challenge-14/challenge-14.js

```diff
@@ -1,97 +1,137 @@
-/*
-Envolva todo o código desse desafio em uma IIFE.
-*/
+(function(){
+	/*
+	Envolva todo o código desse desafio em uma IIFE.
+	*/
 
-/*
-Crie um array chamado numberObjects. Esse array deve ter 10 elementos. Cada
-elemento será um objeto no formato:
-{ number: [NUMBER] }
-Os números devem ser de 1 a 10.
-Mostre esse array no console.
-*/
-console.log( 'Number Objects Array:' );
-// ?
+	/*
+	Crie um array chamado numberObjects. Esse array deve ter 10 elementos. Cada
+	elemento será um objeto no formato:
+	{ number: [NUMBER] }
+	Os números devem ser de 1 a 10.
+	Mostre esse array no console.
+	*/
+	console.log( 'Number Objects Array:' );
+	var numberObjects = [];
+	for( var i = 0; i <= 10; i++ ){
+		numberObjects.push({ number: i })
+	}
 
-/*
-Crie um array chamado `justNumbers`, que terá como elementos somente os
-números do array criado acima. Mostre esse novo array no console.
-*/
-console.log( '\nJust Numbers:' );
-// ?
+	console.log(numberObjects);
 
-/*
-Crie um novo array chamado `justMod2Or3`, que receberá do array criado acima
-somente os números que forem divisíveis por 2 ou 3. Mostre esse novo array
-no console.
-*/
-console.log( '\nJust module of division by 2 or 3:' );
-// ?
+	/*
+	Crie um array chamado `justNumbers`, que terá como elementos somente os
+	números do array criado acima. Mostre esse novo array no console.
+	*/
+	console.log( '\nJust Numbers:' );
+	var justNumbers = numberObjects.map(function(item){
+		return item.number;
+	});
+	console.log(justNumbers);
 
-/*
-Declare uma variável chamada operation que receba, do array criado acima,
-um valor reduzido pela seguinte operação:
-- Somar 1 ao último valor retornado;
-- Multiplicar o resultado pelo valor atual.
-O cálculo deve começar com zero.
-Mostre o resultado no console.
-*/
-console.log( '\nOperation:' );
-// ?
+	/*
+	Crie um novo array chamado `justMod2Or3`, que receberá do array criado acima
+	somente os números que forem divisíveis por 2 ou 3. Mostre esse novo array
+	no console.
+	*/
+	console.log( '\nJust module of division by 2 or 3:' );
+	var justMod2Or3 = justNumbers.filter(function(item){
+		if( item % 2 === 0 || item % 3 === 0 ){
+			return item;
```

Comment:
-1
Com o `filter` você não precisa retornar o `item`, pois o método já faz isso por você.

Você deve retornar uma expressão: se ela for verdadeira, o `filter` retorna o `item`. Caso contrário, ele passa para o próximo :)